File: SDKs/amazonaws.com/rds/2014-10-31/go/pkg/models/shared/eventlist.go
package shared

import (
	"time"
)

// EventList
//
//	This data type is used as a response element in the <code>DescribeEvents</code> action.
type EventList struct {
	Date             *time.Time
	EventCategories  []map[string]interface{}
	Message          *string
	SourceArn        *string
	SourceIdentifier *string
	SourceType       *SourceTypeEnum
}

File: SDKs/amazonaws.com/rds/2014-10-31/typescript/src/sdk/models/shared/eventlist.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { SourceTypeEnum } from "./sourcetypeenum";



// EventList
/** 
 *  This data type is used as a response element in the <code>DescribeEvents</code> action. 
**/
export class EventList extends SpeakeasyBase {
  @SpeakeasyMetadata()
  date?: Date;

  @SpeakeasyMetadata()
  eventCategories?: Record<string, any>[];

  @SpeakeasyMetadata()
  message?: string;

  @SpeakeasyMetadata()
  sourceArn?: string;

  @SpeakeasyMetadata()
  sourceIdentifier?: string;

  @SpeakeasyMetadata()
  sourceType?: SourceTypeEnum;
}
